src/component/feature/FeatureStrategy/FeatureStrategyContainer/FeatureStrategyContainer.tsx

```diff
@@ -0,0 +1,140 @@
+    const StrategyIcon = getFeatureStrategyIcon(strategy.name ?? '');
+    const strategyName = getHumanReadableStrategyName(strategy.name ?? '');
+    const { uiConfig } = useUiConfig();
+    const { push } = useHistory();
+
+    const onCancel = () => {
+        push(formatFeaturePath(feature.project, feature.name));
+    };
+
+    const onSubmitOrProdGuard = async (event: React.FormEvent) => {
+        event.preventDefault();
+        if (enableProdGuard) {
+            setShowProdGuard(true);
+        } else {
+            onSubmit();
+        }
+    };
+
+    // TODO(olav): Remove FeatureStrategyConstraints when the new constraints are released.
+    const FeatureStrategyConstraintsImplementation = uiConfig.flags.CO
+        ? FeatureStrategyConstraints2
+        : FeatureStrategyConstraints;
+
+    return (
+        <form className={styles.form} onSubmit={onSubmitOrProdGuard}>
+            <h2 className={styles.title}>
+                <StrategyIcon className={styles.icon} aria-hidden />
+                <span className={styles.name}>{strategyName}</span>
+            </h2>
+            <div>
+                <FeatureStrategyEnabled
+                    feature={feature}
+                    environmentId={environmentId}
+                />
+            </div>
+            <ConditionallyRender
+                condition={Boolean(uiConfig.flags[C])}
+                show={
+                    <div>
+                        <FeatureStrategyConstraintsImplementation
+                            projectId={feature.project}
+                            environmentId={environmentId}
+                            strategy={strategy}
+                            setStrategy={setStrategy}
+                        />
+                    </div>
+                }
+            />
+            <FeatureStrategyType
+                strategy={strategy}
+                setStrategy={setStrategy}
+                editable={editable}
+            />
+            <div className={styles.buttons}>
+                <Button
+                    type="button"
+                    color="secondary"
+                    onClick={onCancel}
+                    disabled={loading}
+                >
+                    Cancel
+                </Button>
+                {strategy.id && (
+                    <FeatureStrategyRemove
+                        projectId={feature.project}
+                        featureId={feature.name}
+                        environmentId={environmentId}
+                        strategyId={strategy.id}
+                        disabled={loading}
+                    />
+                )}
+                <FeatureStrategyProdGuard
+                    open={showProdGuard}
+                    onClose={() => setShowProdGuard(false)}
+                    onClick={onSubmit}
+                    loading={loading}
+                    label="Save strategy"
+                />
+                <Button
```

Comment:
Having the buttons here places them directly underneath the form instead of at the bottom like FormTemplate provider does. It would be good if we could use the same location so that if we decide to do something with the buttons (for example make them floating) we'd only have to work with the container in the FormTemplate.